src/NzbDrone.Core/DecisionEngine/Specifications/QueueSpecification.cs

```diff
@@ -42,6 +43,16 @@ public Decision IsSatisfiedBy(RemoteEpisode subject, SearchCriteriaBase searchCr
                 var qualityProfile = subject.Series.QualityProfile.Value;
                 var languageProfile = subject.Series.LanguageProfile.Value;
 
+                // If the item has already failed ignore it, it'll be cleaned up automatically later.
+                // To avoid a race make it's it's not FailedPending (failed awaiting removal/search)
+                // or Failed (removed/searching) though the latter should be removed automatically.
+
+                if (queueItem.TrackedDownloadState == TrackedDownloadState.FailedPending ||
+                    queueItem.TrackedDownloadState == TrackedDownloadState.Failed)
```

Comment:
'Failed' doesn't happen since DownloadIsTrackable function already rejects those. 
FailedPending are listed, but I don't think that this check is needed.

Reply:
This is is for https://github.com/Sonarr/Sonarr/issues/2973 it's rare, but does happen.

Reply:
Failed items never appear in the queue coz you actively filter those out here https://github.com/Sonarr/Sonarr/pull/3262/files#diff-5ab3194831f90f6cba9326ea9dd65a2aR174
Thing is that the queue is a copy, so when you set TrackedDownload.State == FailedPending, those items will still be as Downloading in the queue till he event is raised (https://github.com/Sonarr/Sonarr/pull/3262/files#diff-5ab3194831f90f6cba9326ea9dd65a2aR76).
In ProcessFailed you set the Failed state, but that isn't propagated yet either. You immediately raise the (async) event that'll lead to the search.
Only at the end of ProcessMonitoredDownloadsCommand  you propagate those states to the queue.
So yes, it's very likely that the queue is updated before the search finishes, but it isn't guaranteed.

I think the QueueSpecification should be 'Temporary', this would cause those releases to be added to the pending/fallback queue instead of being dropped entirely. Then the timing doesn't matter much anymore since the next rsssync would pick them up.

Reply:
`FailedPending` is applied to the queue when https://github.com/Sonarr/Sonarr/pull/3262/files#diff-5ab3194831f90f6cba9326ea9dd65a2aL75 executes (and the queue updates). That happens before the items are removed from the queue entirely (and replacements searched for) and happens synchronously so the search doesn't happen before the internal queue state is applied.

You're right about `Failed` not making it this far though.

> I think the QueueSpecification should be 'Temporary'

I can see how it'd help here, but worried that doing so would result in something being removed from the queue manually by the user and suddenly another release shows up, maybe it's not an issue with our history spec. I won't worry about it here, but we can discuss and decide one way or another.